resources/FindWebElements.py

```diff
@@ -148,29 +103,237 @@ def filter_elements_by_attributes_by_driver(self, scanned_element_path:str, attr
+
+        Retourne:
+        - Nombre d'éléments correspondants dans element_siblings.
+        """
+        count = 0
+        for sibling, prop in zip(element_siblings, siblings_properties):
+            # Vérifie les correspondances des propriétés
+            if (
+                sibling.tag_name == prop["tagName"] and
+                sibling.text.strip() == prop["textContent"]
+            ):
+                count += 1
+        return count
 
-        return dom_elements
     
+    def find_best_element_with_siblings(self, attributes_way_result, siblings):
+        """
+        Trouve l'élément de attributes_way_result ayant le plus de siblings correspondant
+        aux propriétés spécifiées dans la liste siblings.
+
+        Arguments:
+        - attributes_way_result: Liste d'éléments WebElement trouvés par Selenium.
+        - siblings: Liste de dictionnaires représentant les propriétés des siblings d'un élément.
+
+        Retourne:
+        - L'élément WebElement de attributes_way_result ayant le plus de correspondances avec siblings.
+        """
+
+        # Initialisation des variables pour suivre le meilleur élément
+        best_element = None
+        max_matches = 0
+        elem = 0
+
+        # Parcours de chaque élément pour trouver celui avec le maximum de correspondances
+        for element in attributes_way_result:
+            # Obtenez tous les siblings de l'élément
+            element_siblings = element.find_elements(By.XPATH, "following-sibling::*")
+            
+            # Appel de match_count pour calculer le nombre de correspondances avec siblings
+            matches = self.match_count(element_siblings, siblings)
+            elem = elem + 1
+            print(f"Elements {elem} a {matches} frères en commun avec l'élément cherché")
+
+            # Met à jour le meilleur élément si un nombre plus élevé de correspondances est trouvé
+            if matches > max_matches:
+                max_matches = matches
+                best_element = element
+
+        return best_element
+
+
+    # Fonction pour faire clignoter l'élément
+    def blink_element(self, element, duration=3, interval=0.3):
+        selenium_lib = BuiltIn().get_library_instance('SeleniumLibrary')
+        # Faire défiler pour amener l'élément dans la vue
+        selenium_lib.driver.execute_script("arguments[0].scrollIntoView({behavior: 'smooth', block: 'center'});", element)
+        time.sleep(0.5)  # Pause pour s'assurer que le défilement est terminé
+
+        end_time = time.time() + duration
+        while time.time() < end_time:
+            # Changer la couleur de fond en jaune
+            selenium_lib.driver.execute_script("arguments[0].style.backgroundColor = 'blue'", element)
+            time.sleep(interval)
+            
+            # Remettre la couleur de fond d'origine
+            selenium_lib.driver.execute_script("arguments[0].style.backgroundColor = ''", element)
```

Comment:
pas sûr que remettre le backgroundColor a 'rien' remette la couleur d'origine.
il faudrait stocker ça valeur de base pour la remettre par la suite

Reply:
Ajout des lignes :
        # Retrieve the original background color of the element
        original_bg_color = selenium_lib.driver.execute_script("return arguments[0].style.backgroundColor;", element)